front/pages/api/apps/[user]/[sId]/datasets/[name]/[hash]/index.ts

```diff
@@ -1,22 +1,52 @@
-import { NextApiRequest, NextApiResponse } from "next";
 import { auth_user } from "@app/lib/auth";
-import { User, App, Dataset } from "@app/lib/models";
+import { DustAPI, isErrorResponse } from "@app/lib/dust_api";
+import { parse_payload } from "@app/lib/http_utils";
+import { App, Dataset, User } from "@app/lib/models";
 import withLogging from "@app/logger/withlogging";
 import { DatasetType } from "@app/types/dataset";
+import { JSONSchemaType } from "ajv";
+import { NextApiRequest, NextApiResponse } from "next";
 
 const { DUST_API } = process.env;
 
 type GetDatasetByHashResponseBody = { dataset: DatasetType };
 
+type GetDatasetQuery = {
+  hash: string;
+  user: string;
+  sId: string;
+  name: string;
+};
+
+const getDatasetQuerySchema: JSONSchemaType<GetDatasetQuery> = {
+  type: "object",
+  properties: {
+    hash: { type: "string" },
+    user: { type: "string" },
+    sId: { type: "string" },
+    name: { type: "string" },
+  },
+  required: ["hash", "user", "sId", "name"],
+};
+
 async function handler(
   req: NextApiRequest,
   res: NextApiResponse<GetDatasetByHashResponseBody>
 ): Promise<void> {
+  const getDatasetQueryRes = parse_payload(getDatasetQuerySchema, req.query);
```

Comment:
let's call it queryRes and query to be a bit less verbose?

Reply:
i copied from search for consistency, I can change it over there as well